core/src/main/scala/org/apache/spark/storage/DiskBlockObjectWriter.scala

```diff
@@ -80,12 +83,28 @@ private[spark] class DiskBlockObjectWriter(
    */
   private var numRecordsWritten = 0
 
+  private var sparkConf: SparkConf = null
+
   def open(): DiskBlockObjectWriter = {
     if (hasBeenClosed) {
       throw new IllegalStateException("Writer already closed. Cannot be reopened.")
     }
     fos = new FileOutputStream(file, true)
-    ts = new TimeTrackingOutputStream(writeMetrics, fos)
+    if (CryptoConf.isShuffleEncryted(sparkConf)) {
+      val cryptoCodec: CryptoCodec = CryptoCodec.getInstance(sparkConf)
```

Comment:
It feels like this code (and the respective code in that creates the input stream) should be in a helper method somewhere. That should make it easier to test `CryptoInputStream` and `CryptoOutputStream` in isolation.